### DISCUSSION TITLE
Header-Properties for HeaderedControls

### DISCUSSION THREAD

**Discussion by timunie** [AvaloniaUI/Avalonia#9273] [Polls]
Currently it's sometimes unclear where properties like `Padding`, `Background`, `Foreground` etc. should be applied to. The header-part or the content-part? This affects all `HeaderedContentControls` like `TabControl`, `Expander`, ... See https://github.com/AvaloniaUI/Avalonia/pull/9270 as example

**Comment by robloo** [comment]
This was already well debated in WinUI. The consensus -- and the way WinUI works now-- is that the properties need to follow the Content property which is specially handled in XAML.

If I set the content of an Expander and it goes below the header in the content area why would I expect setting Padding would actually apply to the Header the way it currently works? Padding is conceptually to surround the content.

So the Padding property needs to go with the Content. Then what to do about Background, BorderBrush, etc... Well, we dare not make some properties apply to the header and some to the content area. By simple extension these need to apply to the content area as well.

If I have the time I'll add links to the old discussions in WinUI. The arguments are pretty logical though as stated above.

**Reply by timunie** [reply]
I fully agree with you @robloo. The question where I'm not 100 % sure is about additional properties for the Header. We had them in MahApps and they were useful. But as Avalonias Styling is way more flexible than WPFs, it may not be worth the effort. This was the main intention for this poll. I hope you can follow my thoughts 💭.

**Comment by robloo** [comment]
> Padding, Backgrond, Foregroung all apply to Content. Header is just styleable via ControlTemplate

This isn't entirely true. There are now resources to control all of this with the lightweight styling concept. There is absolutely no need to re-template to set things like header background.

**Reply by timunie** [reply]
I haven't used this yet heavily. You may be entirely true here, as you did a great job on color picker and other controls. 👍

**Reply by robloo** [reply]
I suppose this is theme dependent. If anyone is using Fluent theme in Avalonia (Fluent v1) or Fluent v2 from FluentAvalonia, they are designed to have resources that can be used for lightweight styling. Other themes not originating from Microsoft probably don't follow this pattern.

**Comment by maxkatz6** [comment]
Another option, maybe debatable, is to add `Expander.HeaderTheme` property, so it can be styled with more flexibility than `Header***` properties. But at the same time slightly easier than using deep selectors.
```
<Expander>
   <Expander.HeaderTheme>
       <ControlTheme BasedOn="{StaticResource FluentExpanderHeaderTheme}">
           <Setter ... />
       </ControlTheme>
   </Expander.HeaderTheme>
</Expander>
```

**Reply by timunie** [reply]
Hey @maxkatz6 I think this is a really great idea. I updated the poll to reflect this. Also I directly voted for it.

**Reply by robloo** [reply]
I'm not so sure, in the case of Expander you would have to switch out the entire toggle button control theme that makes up the header. The toggle button header isn't simple like an items control and has a background, content area, a chevron area, animations and several pseudoclass states. You would be re-templating most of the control to do this. 

It might conceptually match with other controls, however I wouldn't force devs to use this to change the header of an Expander for example. So basically, I would consider it a backup option but definitely not better than having lightweight resources or even new header properties.

**Comment by rabbitism** [comment]
It's quite hard to determine, because different controls have different focus. 
For HeaderedContentControl, like Expander, Content is my focus. However for HeaderedItemsControl, like MenuItem, then Header is my focus.

**Reply by timunie** [reply]
I may have pointed this out better. The poll is about headered content control.
